site/app/controllers/admin/PlagiarismController.php

```diff
@@ -200,96 +209,111 @@ private function saveNewProvidedCode(string $temporary_file_path, string $filena
+        //     foreach ($gradeables_with_plagiarism_result as $gradeable_id_title) {
+        //         $nightly_rerun_info[$gradeable_id_title['g_id']] = false;
+        //     }
+        //     if (file_put_contents($nightly_rerun_info_file, json_encode($nightly_rerun_info, JSON_PRETTY_PRINT)) === false) {
+        //         die("Failed to create nightly rerun info file");
+        //     }
+        // }
+        // else {
+        //     $nightly_rerun_info = json_decode(file_get_contents($nightly_rerun_info_file), true);
+        //     foreach ($nightly_rerun_info as $gradeable_id => $nightly_rerun_status) {
+        //         $flag = 0;
+        //         foreach ($gradeables_with_plagiarism_result as $gradeable_id_title) {
+        //             if ($gradeable_id_title['g_id'] == $gradeable_id) {
+        //                 $flag = 1;
+        //                 break;
+        //             }
+        //         }
+        //         if ($flag == 0) {
+        //             #implies plagiarism result for this gradeable are deleted
+        //             unset($nightly_rerun_info[$gradeable_id]);
+        //         }
+        //     }
+        //
+        //     foreach ($gradeables_with_plagiarism_result as $gradeable_id_title) {
+        //         if (!array_key_exists($gradeable_id_title['g_id'], $nightly_rerun_info)) {
+        //             #implies plagiarism was run for this gradeable
+        //             $nightly_rerun_info[$gradeable_id_title['g_id']] = false;
+        //         }
+        //     }
+        //     if (file_put_contents($nightly_rerun_info_file, json_encode($nightly_rerun_info, JSON_PRETTY_PRINT)) === false) {
+        //         die("Failed to create nightly rerun info file");
+        //     }
+        // }
+        $nightly_rerun_info = []; // placeholder
+
+        $this->core->getOutput()->renderOutput(['admin', 'Plagiarism'], 'plagiarismMainPage', $all_configurations, $refresh_page, $nightly_rerun_info);
     }
 
     /**
+     * @param string $gradeable_id
+     * @param string $config_id
+     *
      * @Route("/courses/{_semester}/{_course}/plagiarism/gradeable/{gradeable_id}")
      */
-    public function showPlagiarismResult($gradeable_id) {
-        $semester = $this->core->getConfig()->getSemester();
-        $course = $this->core->getConfig()->getCourse();
+    public function showPlagiarismResult(string $gradeable_id, string $config_id) {
         $gradeable_config = $this->core->getQueries()->getGradeableConfig($gradeable_id);
         $gradeable_title = $gradeable_config->getTitle();
 
-        $rankings = $this->getOverallRankings($gradeable_id);
+        try {
+            $this->verifyGradeableAndConfigAreValid($gradeable_id, $config_id);
+        }
+        catch (\Exception $e) {
+            $return_url = $this->core->buildCourseUrl(['plagiarism']);
+            $this->core->addErrorMessage($e);
+            $this->core->redirect($return_url);
```

Comment:
If you switch to using WebResponses you could just return a RedirectResponse here.